src/transformStateWithClones.js

```diff
@@ -7,7 +7,51 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let lastState = state;
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties': {
+        const newState = {
+          ...lastState,
+          ...extraData,
+        };
```

Comment:
you can do this copy on the 11 line and don't rewrite it again inside the each case)